### DISCUSSION TITLE
RoamResearch as headless CMS

### DISCUSSION THREAD

**Discussion by odyslam** [roam-garden/i#57] [Q&A]
Hey,

Coming here from the tweet from Conaw, I like this idea a lot. In essence, what I would like from Roam is to be able to automatically create a page from every RR page and correctly cross-link them.

What I have in mind is to create a blog, where the blog posts that I publish are actually RR pages. I would like to embed this into a larger website, so I want to build the front-end, using RR only for the blog content. What I imagine is that every time I create a new public webpage in RR, it calls a worker which gets the page's data, creates the appropriate .md file and kicks off a static website build flow.

It would need to convert all [[]] into []() links for the website and maybe define a couple of "rules" in order to convert the bullet point format into a sanely formatted text (e.g every top-level bullet is a paragraph, etc.)

What do you think folks?

**Comment by Stvad** [comment]
Hey @OdysLam,

I assume you've seen https://roam.garden/ ? 🙂 (which this forum is about 😛)
It takes a more holistic aproach (esp with #28 ) of driving the full pipeline from Roam to a static site. It does resolve the wikilinks and other Roam features (block refs, embeds, custom MD syntax, etc). 

So basically you press a button in Roam and get static site on the other end (there is no "publish on change" behaviour yet, but it should be reasonably easy to add)

I'm not quite sure what you mean by "embed this into a larger website" so depending on what level of integration you envisioning that may or may not work for you.   
Please share your impressions if you give it a try/let me know if it works for you 🙂
